include/ozo/impl/read.h

```diff
@@ -23,10 +23,11 @@ inline Require<RawDataWritable<T>, istream&> read(istream& in, T& out) {
 
 template <typename T>
 inline Require<Integral<T> && sizeof(T) == 1, istream&> read(istream& in, T& out) {
-    out = in.get();
-    if (!in) {
+    istream::traits_type::int_type c = in.get();
+    if (!in && istream::traits_type::not_eof(c)) {
```

Comment:
EOF checking is not needed, `if (!in)` is enough. In other cases the functionality is broken - see the test that fails. So it looks like the best solution is:

```c++
in.get(out);
if (!in) {
    throw system_error(error::unexpected_eof);
}
return in;
```

Reply:
Now all of the tests pass.

Did you still want the check for EOF removed?

Reply:
Oops, I didn't figure out that the problem is the negation absence. Well, for now, I'd ask you to remove `EOF` checking, since it may change the throwing exception behaviour, which in case of an additional `EOF` checking depends on [std::basic_ios::exceptions()](https://en.cppreference.com/w/cpp/io/basic_ios/exceptions) method. It looks like a test case should be added to enforce such behavior.

It is clear that such behavior may hide the initial problem with the stream. For that case, the solution which throws a proper exception is needed. And IMO this is a subject for other PR.

Reply:
Ok. I've removed the check for `not_eof(c)`